helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/provisioning/HelpDeskProvisioningService.java

```diff
@@ -0,0 +1,99 @@
+package org.symphonyoss.symphony.bots.helpdesk.bot.provisioning;
+
+import com.gs.ti.wpt.lc.security.cryptolib.PBKDF;
+
+import com.google.common.io.BaseEncoding;
+import org.glassfish.jersey.internal.util.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.symphonyoss.symphony.bots.helpdesk.bot.client.HelpDeskPublicApiClient;
+import org.symphonyoss.symphony.bots.helpdesk.bot.config.ProvisioningConfig;
+import org.symphonyoss.symphony.bots.helpdesk.bot.util.CertificateUtils;
+import org.symphonyoss.symphony.pod.model.CompanyCert;
+import org.symphonyoss.symphony.pod.model.CompanyCertStatus;
+
+import java.io.File;
+import java.security.cert.X509Certificate;
+
+/**
+ * Performs the provisioning process of a bot. It consists of:
+ *
+ * - Generate a self-signed certificate and submit it to a POD;
+ *
+ * Created by campidelli on 3/9/18.
+ */
+@Service
+public class HelpDeskProvisioningService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HelpDeskProvisioningService.class);
+
+  @Autowired
+  ProvisioningConfig config;
```

Comment:
You might define all fields as private fields.

Reply:
Done.